src/librustdoc/html/markdown.rs

```diff
@@ -947,15 +1240,35 @@ impl LangString {
                         None
                     } {
                         if let Some(extra) = extra {
-                            extra.error_invalid_codeblock_attr(
+                            extra.error_invalid_codeblock_attr_with_help(
                                 &format!("unknown attribute `{}`. Did you mean `{}`?", x, flag),
                                 help,
                             );
                         }
                     }
                     seen_other_tags = true;
+                    data.unknown.push(x.to_owned());
+                }
+                LangStringToken::LangToken(x) => {
+                    seen_other_tags = true;
+                    data.unknown.push(x.to_owned());
+                }
+                LangStringToken::KeyValueAttribute(key, value) => {
+                    seen_other_tags = true;
+                    if key != "class" {
+                        if let Some(extra) = extra {
+                            extra.error_invalid_codeblock_attr(&format!(
+                                "unsupported attribute `{key}`"
+                            ));
+                        }
+                    } else {
+                        data.added_classes.push(value.to_owned());
+                    }
```

Comment:
I would prefer using `==` in the condition to have the shorter branch first, IMO it makes reading clearer when the `else` branch is as close as possible to its condition

Reply:
It also allows to directly have `else if let Some(extra) = extra {`, so much better indeed!